### DISCUSSION TITLE
the marker is not deleted from the map

### DISCUSSION THREAD

**Discussion by invweb** [osmdroid/osmdroid#1978] [Q&A] [answered]
There is such a code:
```
 val markerDeletionClickListener = object : MarkerRemover {
            override fun removeMarker() {
                mapView?.overlays?.remove(marker) //to delete marker from MapView
                mapView?.invalidate()
            }
        }
```
The code should cause the marker to be removed from the map, but this does not happen. What should I do?

**Comment by invweb** [comment]
deletion occurs for one marker out of many, the rest are not deleting.

**Comment by invweb** [comment]
This code is applied to many markers, but occurs at one marker out of many, the rest are not deleted.

**Reply by MuzSumer** [reply, marked as answer]
Markers are stored in sqllite database. Every map that has bookmark overlay shows the same markers.

See the OSM Map sample code to use itemized overlay instead.
